docs/examples/custom_destination_lancedb/custom_destination_lancedb.py

```diff
@@ -0,0 +1,167 @@
+
+
+def fetch_show_episode_data(
+    show_id: str, access_token: Optional[str] = None, params: Dict[str, Any] = None
+):
+    """Fetch all shows data from Spotify API based on endpoint and params."""
+    url = f"{BASE_SPOTIFY_URL}/shows/{show_id}/episodes"
+    if params is None:
+        params = {}
+    headers = {"Authorization": f"Bearer {access_token}"} if access_token else {}
+    while True:
+        response_ = requests.get(url, params=params, headers=headers)
+        response_.raise_for_status()
+        response = response_.json()
+        yield response["items"]
+        if not response or "next" not in response or response["next"] is None:
+            break
+        url = response["next"]
+
+
+@dlt.source
+def spotify_shows(client_id: str = dlt.secrets.value, client_secret: str = dlt.secrets.value):
+    access_token: str = get_spotify_access_token(client_id, client_secret)
+    params: Dict[str, Any] = {"limit": 50}
+    for show in fields(Shows):
+        show_name: str = show.name
+        show_id: str = show.default  # type: ignore[assignment]
+        yield dlt.resource(
+            fetch_show_episode_data(show_id, access_token, params),
+            name=show_name,
+            write_disposition="merge",
+            primary_key="id",
+            parallelized=True,
+            max_table_nesting=0,
+        )
+
+
+@dlt.destination(batch_size=250, name="lancedb")
+def lancedb_destination(items: TDataItems, table: TTableSchema) -> None:
+    db = lancedb.connect(db_path, read_consistency_interval=datetime.timedelta(0))
+    try:
+        tbl = db.open_table(table["name"])
+    except FileNotFoundError:
+        tbl = db.create_table(table["name"], schema=EpisodeSchema)
+    tbl.checkout_latest()
+    tbl.add(items)
+
+
+if __name__ == "__main__":
+    db = lancedb.connect(db_path, read_consistency_interval=datetime.timedelta(0))
+
+    for show in fields(Shows):
+        db.drop_table(show.name, ignore_missing=True)
+
+    pipeline = dlt.pipeline(
+        pipeline_name="spotify",
+        destination=lancedb_destination,
+        dataset_name="spotify_podcast_data",
+        progress="log",
+    )
+
+    load_info = pipeline.run(
+        spotify_shows(client_id=dlt.secrets.value, client_secret=dlt.secrets.value)
+    )
+
+    row_counts = pipeline.last_trace.last_normalize_info
+
+    print(row_counts)
+    print("------")
+
+    print(load_info)
+
+    print("------")
+    print("------")
+
+    # Showcase vector search capabilities over our dataset with lancedb.
+    # Perform brute force search while we have small data.
+    query = "French AI scientist with Lex, talking about AGI and Meta and Llama"
+    table_to_query = "lex_fridman"
+
+    print(f"Query: {query}")
+    print(f"Querying table: {table_to_query}")
+
+    tbl = db.open_table(table_to_query)
+    tbl.checkout_latest()
+
+    results = tbl.search(query=query).to_list()
+    print(results)
```

Comment:
maybe add an assertion here, it will be useful for the test

Reply:
added!